### ISSUE TITLE
About Regularization Image

### ISSUE THREAD

**Issue by atodniAr** [kohya-ss/sd-scripts#330]
There are two things I'm not very clear about reg image in DreamBooth and LoRA training.

1. The current implementation will discard parts of reg image if it exceeds the amount of train image (after repeat). Is there a specific reason for this limitation? Since I check the original DreamBooth paper, they used 200x amout of reg images. Theoretically you should use more image to better approximate the prior class distribution.
2. If you provided caption files for all images (including tran and reg) then the caption will be used directly for text conditioning vector generation right? I read through the code but am not sure if I missed something. If that's the case, there is another problem when you use folder name but not text caption. E.g. if you used "10_aaa 1girl" for both train and reg images, then "aaa 1girl" will be used for both train and reg subset. However the correct way to text conditioning should be "aaa 1girl" for train images and "1girl" for reg images.

I'm not sure if any of these two problems really exists since I had a hard time read through the code, especially for the dataset pipeline part. I cound miss some important code. If that's the case please let me know :)

**Comment by AI-Casanova** [comment]
1) Can be avoided by passing reg images as train images, as I read the code, the only difference in handling between the two (besides the limit) is the ability to pass a loss factor other that 1 to a reg image. 

2) is simply a matter of correctly labelling your directories, or using a TOML config with `class_token =`

**Comment by atodniAr** [comment]
> 1. Can be avoided by passing reg images as train images, as I read the code, the only difference in handling between the two (besides the limit) is the ability to pass a loss factor other that 1 to a reg image.
> 2. is simply a matter of correctly labelling your directories, or using a TOML config with `class_token =`

Yes, you can make such workaround. However if that's the case, it's better to do an easy fix to prevent confusion. I actually had to read source code here only for making sure the reg image works just as stated in the original DreamBooth paper.

**Comment by hello-navi** [comment]
When you use keep_tokens, are they being kept (unshuffled) for reg images as well?

**Comment by AI-Casanova** [comment]
> When you use keep_tokens, are they being kept (unshuffled) for reg images as well?

If you search `train_utils.py` for `is_reg` the only downstream use is for `loss_weights`

In other words, once reg images are loaded, they should be handled exactly like other images, apart from the aforementioned `loss_weights`

**Comment by AI-Casanova** [comment]
> When you use keep_tokens, are they being kept (unshuffled) for reg images as well?

If you search `train_utils.py` for `is_reg` the only downstream use is for `loss_weights`

In other words, once reg images are loaded, they should be handled exactly like other images, apart from the aforementioned `loss_weights`
